joomla/joomla-cms
# FIX #37772 - inlineHelp Toggle Button control

Pull Request for Issue #37772 .

### Summary of Changes
- Add inlineHelp option to formField
- Control if inlineHelp toggle button is enabled for the current form
- Add 'hide-aware-inline-help d-none' if enabled, as default description class
- ~~You can know add an xml form field attribute "inlinehelp" to use or not the toggling of description field by field (values boolean true/false). Eg. inlinehelp="false" applied on one field, will keep always the description shown even if click on description toggle.~~

UPDATE: see PR #37849 removing the inlinehelp attribute availability for form field xml element.



### Testing Instructions
- Check if descriptions are hidden by default in com_config and com_content options.
- Check if descriptions are shown by default everywhere else (where toggle button is not integrated) : component options, item edition, plugins manager... and in third-party extensions not integrating the toggle button.
- Toggle button works as expected in com_config and com_content



### Actual result BEFORE applying this Pull Request
- Description is hidden on all places where inlinehelp toggle button is not integrated. Almost everywhere in 4.1.3.


### Expected result AFTER applying this Pull Request
- Description is shown if no inlinehelp toggle button.


### Documentation Changes Required
inline help global documentation, and form field attribute inlinehelp.
